gravitee-apim-console-webui/src/services-ngx/constants.service.ts

```diff
@@ -93,6 +93,12 @@ export class ConstantsService {
       return availablePlanMenuItems.filter((planMenuItem) => planMenuItem.planFormType !== 'PUSH');
     }
 
+    if (definitionVersion === 'V4' && listenerTypes?.every((listenerType) => listenerType === 'KAFKA')) {
+      return availablePlanMenuItems
+        .filter((planMenuItem) => planMenuItem.planFormType !== 'PUSH')
+        .filter((planMenuItem) => planMenuItem.planFormType !== 'MTLS');
```

Comment:
We'll have to think about adding this to the rest-api side as well, with this task https://gravitee.atlassian.net/browse/APIM-7752